src/Controllers/AnnouncementController.php

```diff
@@ -104,4 +107,78 @@ public function edit(Request $request, $id)
             return ApiResponse::failed($e);
         }
     }
+
+    public function delete($id)
+    {
+        try {
+            $user = auth()->user();
+
+            $announcement = Announcement::where('id', $id)
+                ->first();
+
+            if (! $announcement) {
+                throw ValidationException::withMessages([
+                    'id' => 'Announcement not found',
+                ]);
+            }
+
+            $authorized_announcement = Announcement::where('id', $id)
+                ->where('created_by', $user->id)
+                ->first();
+
+            if (! $authorized_announcement) {
+                $course = Course::where('id', $announcement->course_id)
+                    ->where('created_by', $user->id)
+                    ->first();
+
+                if (! $course) {
+                    throw ValidationException::withMessages([
+                        'id' => 'Announcement not found',
+                    ]);
+                }
+            }
+
+            if (! CourseUserHelper::isUserInCourse($user->id, $announcement->course_id)) {
+                throw ValidationException::withMessages([
+                    'id' => 'Must enroll the course to edit the announcement',
+                ]);
+            }
+
+            $announcement->delete();
+
+            return ApiResponse::success($announcement->toArray());
+        } catch (Exception $e) {
+            return ApiResponse::failed($e);
+        }
+    }
+
+    public function comment(Request $request)
+    {
+        try {
+            $user = auth()->user();
+            $request->validate([
+                'announcement_id' => 'required|integer',
+                'content' => 'required|string|max:65535',
+            ]);
+
+            $announcement = Announcement::where('id', $request->input('announcement_id'))
+                ->first();
+
+            if (! $announcement) {
+                throw ValidationException::withMessages([
+                    'announcement_id' => 'Announcement not found',
+                ]);
+            }
+
+            $comment = Comment::create([
+                'announcement_id' => $request->input('announcement_id'),
+                'content' => $request->input('content'),
+                'created_by' => $user->id,
+            ]);
+
+            return ApiResponse::success($comment->toArray());
+        } catch (Exception $e) {
+            return ApiResponse::failed($e);
+        }
+    }
```

Comment:
I don't really remember whose part is this code, but can you please change the function name to `createComment` or `addComment`, cuz I needed to read the whole code to realize that this is intended to create a comment